In [1]:
import pandas as pd
import numpy as np

In [62]:
df = pd.read_csv('/home/eeshan/Documents/pandas/Handson 1/housing.csv')
df.head()


id             date   price  bedrooms  bathrooms  sqft_living   
0  7129300520  20141013T000000  221900         3       1.00         1180  \
1  6414100192  20141209T000000  538000         3       2.25         2570   
2  5631500400  20150225T000000  180000         2       1.00          770   
3  2487200875  20141209T000000  604000         4       3.00         1960   
4  1954400510  20150218T000000  510000         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement   
0      5650     1.0           0     0  ...      7        1180              0  \
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15   
0      1955             0    98178  47.5112 -122.257           1340  \
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

How many customers are having waterfront? - 163

In [25]:
df['waterfront'].value_counts()


waterfront
0    21450
1      163
Name: count, dtype: int64

Which zip code is having the costliest house? - 98102	

In [42]:
df["price"].sort_values()
df.sort_values(by="price")

id             date    price  bedrooms  bathrooms  sqft_living   
1149   3421079032  20150217T000000    75000         1       0.00          670  \
15293    40000362  20140506T000000    78000         2       1.00          780   
465    8658300340  20140523T000000    80000         1       0.75          430   
16198  3028200080  20150324T000000    81000         2       1.00          730   
8274   3883800011  20141105T000000    82000         3       1.00          860   
...           ...              ...      ...       ...        ...          ...   
1448   8907500070  20150413T000000  5350000         5       5.00         8000   
4411   2470100110  20140804T000000  5570000         5       5.75         9200   
9254   9208900037  20140919T000000  6885000         6       7.75         9890   
3914   9808700762  20140611T000000  7062500         5       4.50        10040   
7252   6762700020  20141013T000000  7700000         6       8.00        12050   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above   
1149      43377     1.0           0     0  ...      3         670  \
15293     16344     1.0           0     0  ...      5         780   
465        5050     1.0           0     0  ...      4         430   
16198      9975     1.0           0     0  ...      5         730   
8274      10426     1.0           0     0  ...      6         860   
...         ...     ...         ...   ...  ...    ...         ...   
1448      23985     2.0           0     4  ...     12        6720   
4411      35069     2.0           0     0  ...     13        6200   
9254      31374     2.0           0     4  ...     13        8860   
3914      37325     2.0           1     2  ...     11        7680   
7252      27600     2.5           0     3  ...     13        8570   

       sqft_basement  yr_built  yr_renovated  zipcode      lat     long   
1149               0      1966             0    98022  47.2638 -121.906  \
15293              0      1942             0    98168  47.4739 -122.280   
465                0      1912             0    98014  47.6499 -121.909   
16198              0      1943             0    98168  47.4808 -122.315   
8274               0      1954             0    98146  47.4987 -122.341   
...              ...       ...           ...      ...      ...      ...   
1448            1280      2009             0    98004  47.6232 -122.220   
4411            3000      2001             0    98039  47.6289 -122.233   
9254            1030      2001             0    98039  47.6305 -122.240   
3914            2360      1940          2001    98004  47.6500 -122.214   
7252            3480      1910          1987    98102  47.6298 -122.323   

       sqft_living15  sqft_lot15  
1149            1160       42882  
15293           1700       10387  
465             1200        7500  
16198            860        9000  
8274            1140       11250  
...              ...         ...  
1448            4600       21750  
4411            3560       24345  
9254            4540       42730  
3914            3930       25449  
7252            3940        8800  

[21613 rows x 21 columns]

In [49]:
#Which zip code is having the costliest house? - 98102	
df.loc[df["price"]==df["price"].max(),"zipcode"]

7252    98102
Name: zipcode, dtype: int64

How many houses are having grade 10? - 1134

In [56]:
df.loc[df["grade"]==10].value_counts()

id          date             price    bedrooms  bathrooms  sqft_living  sqft_lot  floors  waterfront  view  condition  grade  sqft_above  sqft_basement  yr_built  yr_renovated  zipcode  lat      long      sqft_living15  sqft_lot15
11500240    20150428T000000  872750   3         2.50       2870         13695     2.0     0           0     3          10     2870        0              1991      0             98052    47.6944  -122.102  2840           8472          1
5701700640  20140729T000000  849000   3         3.00       2960         42159     2.0     0           0     3          10     2960        0              1995      0             98052    47.7183  -122.100  2640           25209         1
6072800205  20141119T000000  2375000  4         2.50       3220         20251     1.0     0           0     3          10     3220        0              1969      0             98006    47.5692  -122.192  4200           22114         1
6072800170  20150428T000000  2500000  4         4.00       33

How many null values are there in the dataset? - 0

In [61]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  int64  
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

Does this 9126100861 customer have a waterfront? - NO

In [65]:
df.loc[df["id"]==9126100861,"waterfront"]

400    0
Name: waterfront, dtype: int64

How many houses are having 3 views? - 510

In [68]:
df.loc[df['view']==3].value_counts()

id          date             price    bedrooms  bathrooms  sqft_living  sqft_lot  floors  waterfront  view  condition  grade  sqft_above  sqft_basement  yr_built  yr_renovated  zipcode  lat      long      sqft_living15  sqft_lot15
46100204    20150221T000000  1505000  5         3.00       3300         33474     1.0     0           3     3          9      1870        1430           1957      1991          98040    47.5673  -122.210  3836           20953         1
5553300375  20140820T000000  2160000  3         3.50       3080         6495      2.0     0           3     3          11     2530        550            1996      2006          98199    47.6321  -122.393  4120           8620          1
6117500320  20140708T000000  1131000  3         2.25       2790         13791     1.0     0           3     3          8      2790        0              2006      0             98166    47.4389  -122.351  2720           12600         1
6117500160  20150317T000000  425000   3         1.50       15

What is the lowest price of the house? - 75000

In [69]:
df.loc[df["price"]==df["price"].min()]

id             date  price  bedrooms  bathrooms  sqft_living   
1149  3421079032  20150217T000000  75000         1        0.0          670  \

      sqft_lot  floors  waterfront  view  ...  grade  sqft_above   
1149     43377     1.0           0     0  ...      3         670  \

      sqft_basement  yr_built  yr_renovated  zipcode      lat     long   
1149              0      1966             0    98022  47.2638 -121.906  \

      sqft_living15  sqft_lot15  
1149           1160       42882  

[1 rows x 21 columns]

Which zip code area is having the cheapest house? - 98022

In [71]:
df.loc[df["price"]==df["price"].min(),"zipcode"]

1149    98022
Name: zipcode, dtype: int64

Which area is having the biggest sqft_living? - 98053

In [73]:
df.loc[df["sqft_living"]==df['sqft_living'].max(),"zipcode"]

12777    98053
Name: zipcode, dtype: int64

In which year the costliest house has been built? - 2014

In [80]:
df.loc[df["sqft_living"]==df['sqft_living'].max(),"date"]

12777    20140505T000000
Name: date, dtype: object

Which of the following data types can a Pandas Series have?

 a. Int b. Str c. Float d. All of the above

 D

Which of the following is having a one-dimensional array?
a. Series data type b. DataFrame data type c. Both of the above d. None of the above

A

What value you will get on the left side after printing a series format data?
a. Data b. Value c. Index d. All of the above

C

Which of the following is considered as data in pandas?
a. Dictionary b. Boolean c. Nd-array d. All of the above

D

In what format the keys will get converted into when we convert the data into data frame format?
 a. Rows b. Columns c. Indexes d. Records

 C

Write a function to perform the following operation: Increase the grade of the house by 1 if the sqft of the house is greater than 0 and less than equal to 400. If the total sqft of the house is greater than 400, increase the grade of the house by 2. After creating the above function, which syntax will be using when using apply function to run the above created function in the dataset?

In [93]:
def grade_house(house):
    sqft_living,grade = house   
    if(sqft_living > 0 and sqft_living < 400):
        return grade+1
    else:
        return grade+2

df[['sqft_living','grade']].apply(grade_house,axis=1)

0         9
1         9
2         8
3         9
4        10
         ..
21608    10
21609    10
21610     9
21611    10
21612     9
Length: 21613, dtype: int64

How to check the duplicate values on id,grade and location?

df.duplicated(subset=['id','zipcode','grade']) 

In [83]:
df.duplicated(subset=['id','zipcode','grade']) 

0        False
1        False
2        False
3        False
4        False
         ...  
21608    False
21609    False
21610    False
21611    False
21612    False
Length: 21613, dtype: bool

Import the products and customers dataset and perform the left join on both datasets. With the above operation, how many columns are there in the output?

Create a separate data frame that satisfies the conditions below.
1. Houses built before 1980
2. Have more than 2 bedrooms
3. Have more than 2 floors.

In [100]:
df1 = df.loc[(df.yr_built<1980) & (df.bedrooms>2) & (df.floors>2)]
df1

id             date    price  bedrooms  bathrooms  sqft_living   
518    4386700135  20141114T000000  2250000         4       2.25         4760  \
759     770000045  20141024T000000   405600         5       1.50         2830   
960    7578200310  20141112T000000   650000         4       2.00         2208   
1296   5458800125  20140514T000000   925000         4       2.50         2190   
1350   7234600786  20140511T000000   842500         4       2.50         2160   
...           ...              ...      ...       ...        ...          ...   
18535  3667500015  20140925T000000   770000         4       3.50         3680   
18631  1337800665  20140811T000000  1325000         4       3.25         2850   
19068  6613001241  20140811T000000  1415000         4       3.00         3110   
19188  7942601475  20140520T000000   345600         5       3.50         2800   
19272  1337300145  20140721T000000  1800000         4       2.50         3320   

       sqft_lot  floors  waterfront  view  ...  sqft_above  sqft_basement   
518        8036     2.5           0     0  ...        3390           1370  \
759        4000     2.5           0     0  ...        2830              0   
960        5000     3.0           0     0  ...        2208              0   
1296       7350     2.5           0     0  ...        2190              0   
1350       5298     2.5           0     0  ...        2160              0   
...         ...     ...         ...   ...  ...         ...            ...   
18535      2242     2.5           0     0  ...        2670           1010   
18631      4800     2.5           0     0  ...        2700            150   
19068      4408     2.5           0     3  ...        2510            600   
19188      5120     2.5           0     0  ...        2800              0   
19272      8325     2.5           0     0  ...        3320              0   

       yr_built  yr_renovated  zipcode      lat     long  sqft_living15   
518        1916             0    98112  47.6415 -122.285           2950  \
759        1918             0    98118  47.5132 -122.262           1480   
960        1917             0    98116  47.5711 -122.383           1760   
1296       1958             0    98040  47.5786 -122.236           1880   
1350       1902             0    98122  47.6106 -122.310           1720   
...         ...           ...      ...      ...      ...            ...   
18535      1930          2007    98112  47.6192 -122.307           1350   
18631      1905             0    98112  47.6292 -122.312           2850   
19068      1931             0    98105  47.6583 -122.270           3250   
19188      1903          2005    98122  47.6059 -122.310           1780   
19272      1905             0    98112  47.6263 -122.314           3680   

       sqft_lot15  total_sqft  
518          9323       29829  
759          4000       15140  
960          5000       16176  
1296         7350       20960  
1350         2283       13621  
...           ...         ...  
18535        1288       12240  
18631        4800       18150  
19068        5669       19547  
19188        5120       17620  
19272        6050       24695  

[97 rows x 22 columns]

sample_list = [['Carl', 22], ['Martha', 25], ['Calvin', 12], ['Stuart', 15] ] 

convert into df

In [103]:
sample_list = [['Carl', 22], ['Martha', 25], ['Calvin', 12], ['Stuart', 15] ] 

df2 = pd.DataFrame(sample_list, columns=['name','age'])
df2

name  age
0    Carl   22
1  Martha   25
2  Calvin   12
3  Stuart   15

sample_dict = {'Cristiano': ['Ronaldo','Man U', 801], 'Lionel': ['Messi','PSG', 758], 'Luis': ['Suarez','Atletico Madrid', 509], 'Robert': ['Lewandowski','Bayern Munich', 527], 'Zlatan': ['Ibrahimovic','AC Milan',553] }

In [104]:
sample_dict = {'Cristiano': ['Ronaldo','Man U', 801], 'Lionel': ['Messi','PSG', 758], 'Luis': ['Suarez','Atletico Madrid', 509], 'Robert': ['Lewandowski','Bayern Munich', 527], 'Zlatan': ['Ibrahimovic','AC Milan',553] }

df3 = pd.DataFrame(sample_dict)
df3

Cristiano Lionel             Luis         Robert       Zlatan
0   Ronaldo  Messi           Suarez    Lewandowski  Ibrahimovic
1     Man U    PSG  Atletico Madrid  Bayern Munich     AC Milan
2       801    758              509            527          553

sample_tuple = ([1, 'one', 3], [2, 'two', 3], [3, 'Three', 5], [4, 'Four', 4], [5, 'Five', 4])

In [106]:
sample_tuple = ([1, 'one', 3], [2, 'two', 3], [3, 'Three', 5], [4, 'Four', 4], [5, 'Five', 4])
df4 = pd.DataFrame(sample_tuple)
df4

0      1  2
0  1    one  3
1  2    two  3
2  3  Three  5
3  4   Four  4
4  5   Five  4

Create a separate dataframe that contains houses ordered in ascending or descending order of the prices of each house.

In [112]:
df6 = df.sort_values(by="price")
df6

id             date    price  bedrooms  bathrooms  sqft_living   
1149   3421079032  20150217T000000    75000         1       0.00          670  \
15293    40000362  20140506T000000    78000         2       1.00          780   
465    8658300340  20140523T000000    80000         1       0.75          430   
16198  3028200080  20150324T000000    81000         2       1.00          730   
8274   3883800011  20141105T000000    82000         3       1.00          860   
...           ...              ...      ...       ...        ...          ...   
1448   8907500070  20150413T000000  5350000         5       5.00         8000   
4411   2470100110  20140804T000000  5570000         5       5.75         9200   
9254   9208900037  20140919T000000  6885000         6       7.75         9890   
3914   9808700762  20140611T000000  7062500         5       4.50        10040   
7252   6762700020  20141013T000000  7700000         6       8.00        12050   

       sqft_lot  floors  waterfront  view  ...  sqft_above  sqft_basement   
1149      43377     1.0           0     0  ...         670              0  \
15293     16344     1.0           0     0  ...         780              0   
465        5050     1.0           0     0  ...         430              0   
16198      9975     1.0           0     0  ...         730              0   
8274      10426     1.0           0     0  ...         860              0   
...         ...     ...         ...   ...  ...         ...            ...   
1448      23985     2.0           0     4  ...        6720           1280   
4411      35069     2.0           0     0  ...        6200           3000   
9254      31374     2.0           0     4  ...        8860           1030   
3914      37325     2.0           1     2  ...        7680           2360   
7252      27600     2.5           0     3  ...        8570           3480   

       yr_built  yr_renovated  zipcode      lat     long  sqft_living15   
1149       1966             0    98022  47.2638 -121.906           1160  \
15293      1942             0    98168  47.4739 -122.280           1700   
465        1912             0    98014  47.6499 -121.909           1200   
16198      1943             0    98168  47.4808 -122.315            860   
8274       1954             0    98146  47.4987 -122.341           1140   
...         ...           ...      ...      ...      ...            ...   
1448       2009             0    98004  47.6232 -122.220           4600   
4411       2001             0    98039  47.6289 -122.233           3560   
9254       2001             0    98039  47.6305 -122.240           4540   
3914       1940          2001    98004  47.6500 -122.214           3930   
7252       1910          1987    98102  47.6298 -122.323           3940   

       sqft_lot15  total_sqft  
1149        42882       88759  
15293       10387       29991  
465          7500       14610  
16198        9000       21295  
8274        11250       24536  
...           ...         ...  
1448        21750       66335  
4411        24345       81374  
9254        42730       98424  
3914        25449       86784  
7252         8800       64440  

[21613 rows x 22 columns]

 Calculate the mean and standard deviation of all the numerical values in the dataset. For example - the mean for the bedrooms column is 3.370 and the standard deviation is 0.930.

In [117]:
col_list = df.columns
for col in col_list:
    if df[col].dtype=='int':
        print(f'mean of {col} = {df[col].mean().round(0)} \n std of {col} = {df[col].std().round(0)}\n\n')

mean of id = 4580301521.0 
 std of id = 2876565571.0


mean of price = 540088.0 
 std of price = 367127.0


mean of bedrooms = 3.0 
 std of bedrooms = 1.0


mean of sqft_living = 2080.0 
 std of sqft_living = 918.0


mean of sqft_lot = 15107.0 
 std of sqft_lot = 41421.0


mean of waterfront = 0.0 
 std of waterfront = 0.0


mean of view = 0.0 
 std of view = 1.0


mean of condition = 3.0 
 std of condition = 1.0


mean of grade = 8.0 
 std of grade = 1.0


mean of sqft_above = 1788.0 
 std of sqft_above = 828.0


mean of sqft_basement = 292.0 
 std of sqft_basement = 443.0


mean of yr_built = 1971.0 
 std of yr_built = 29.0


mean of yr_renovated = 84.0 
 std of yr_renovated = 402.0


mean of zipcode = 98078.0 
 std of zipcode = 54.0


mean of sqft_living15 = 1987.0 
 std of sqft_living15 = 685.0


mean of sqft_lot15 = 12768.0 
 std of sqft_lot15 = 27304.0


mean of total_sqft = 34022.0 
 std of total_sqft = 64436.0




Perform the following operations on the pandas dataframe.
1. Get the data starting from the rows 25 to 35.
2. Get the price of the houses located at the longitude--122.045 latitude- 47.6168.

In [118]:
df[25:36]

id             date   price  bedrooms  bathrooms  sqft_living   
25  1202000200  20141103T000000  233000         3       2.00         1710  \
26  1794500383  20140626T000000  937000         3       1.75         2450   
27  3303700376  20141201T000000  667000         3       1.00         1400   
28  5101402488  20140624T000000  438000         3       1.75         1520   
29  1873100390  20150302T000000  719000         4       2.50         2570   
30  8562750320  20141110T000000  580500         3       2.50         2320   
31  2426039314  20141201T000000  280000         2       1.50         1190   
32   461000390  20140624T000000  687500         4       1.75         2330   
33  7589200193  20141110T000000  535000         3       1.00         1090   
34  7955080270  20141203T000000  322500         4       2.75         2060   
35  9547205180  20140613T000000  696000         3       2.50         2300   

    sqft_lot  floors  waterfront  view  ...  sqft_above  sqft_basement   
25      4697     1.5           0     0  ...        1710              0  \
26      2691     2.0           0     0  ...        1750            700   
27      1581     1.5           0     0  ...        1400              0   
28      6380     1.0           0     0  ...         790            730   
29      7173     2.0           0     0  ...        2570              0   
30      3980     2.0           0     0  ...        2320              0   
31      1265     3.0           0     0  ...        1190              0   
32      5000     1.5           0     0  ...        1510            820   
33      3000     1.5           0     0  ...        1090              0   
34      6659     1.0           0     0  ...        1280            780   
35      3060     1.5           0     0  ...        1510            790   

    yr_built  yr_renovated  zipcode      lat     long  sqft_living15   
25      1941             0    98002  47.3048 -122.218           1030  \
26      1915             0    98119  47.6386 -122.360           1760   
27      1909             0    98112  47.6221 -122.314           1860   
28      1948             0    98115  47.6950 -122.304           1520   
29      2005             0    98052  47.7073 -122.110           2630   
30      2003             0    98027  47.5391 -122.070           2580   
31      2005             0    98133  47.7274 -122.357           1390   
32      1929             0    98117  47.6823 -122.368           1460   
33      1929             0    98117  47.6889 -122.375           1570   
34      1981             0    98058  47.4276 -122.157           2020   
35      1930          2002    98115  47.6827 -122.310           1590   

    sqft_lot15  total_sqft  
25        4705       13852  
26        3573       12924  
27        3861       10102  
28        6235       17175  
29        6026       20969  
30        3980       15180  
31        1756        6791  
32        5000       16120  
33        5080       11830  
34        8720       21519  
35        3264       12514  

[11 rows x 22 columns]

In [122]:
df.loc[(df['lat']==47.6168) & (df['long']==-122.045)]

id             date   price  bedrooms  bathrooms  sqft_living   
4  1954400510  20150218T000000  510000         3        2.0         1680  \

   sqft_lot  floors  waterfront  view  ...  sqft_above  sqft_basement   
4      8080     1.0           0     0  ...        1680              0  \

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15   
4      1987             0    98074  47.6168 -122.045           1800  \

   sqft_lot15  total_sqft  
4        7503       20743  

[1 rows x 22 columns]

Create a new dataframe after dropping all the columns except, date, price, and floor area

In [134]:
df7 = df[['date','price','total_sqft']]
df7

date   price  total_sqft
0      20141013T000000  221900       15000
1      20141209T000000  538000       21711
2      20150225T000000  180000       22322
3      20141209T000000  604000       15280
4      20150218T000000  510000       20743
...                ...     ...         ...
21608  20140521T000000  360000        7230
21609  20150223T000000  400000       19463
21610  20140623T000000  402101        6417
21611  20150116T000000  400000        8285
21612  20141015T000000  325000        5493

[21613 rows x 3 columns]

 Perform the following operations on the dataframes given below.
A = pd.DataFrame([['Carl', 22],['Martha', 25],['Calvin', 12],['Stuart', 15]], columns=[‘Name’, ‘Age’]) 
B = pd.DataFrame([['Melvin', 25],['Martha', 34],['Lewis', 32],['Leo', 25]], columns=[‘Name’, ‘Age’]) 
1. Left Outer Join
2. Outer Join
3. Inner Join
4. Right Outer Join

In [156]:
A = pd.DataFrame([['Carl', 22],['Martha', 25],['Calvin', 12],['Stuart', 15]], columns=['Name', 'Age']) 
B = pd.DataFrame([['Melvin', 25],['Martha', 34],['Lewis', 32],['Leo', 25]], columns=['Name', 'Age']) 

pd.merge(A,B,on="Age",how='inner')


Name_x  Age  Name_y
0  Martha   25  Melvin
1  Martha   25     Leo

 What will be the correlation between the columns sqft_living and sqft_above?
1. 0.702035 2. 0.754665 3. 0.876597 4. 0.303093

In [141]:
df.head()


id             date   price  bedrooms  bathrooms  sqft_living   
0  7129300520  20141013T000000  221900         3       1.00         1180  \
1  6414100192  20141209T000000  538000         3       2.25         2570   
2  5631500400  20150225T000000  180000         2       1.00          770   
3  2487200875  20141209T000000  604000         4       3.00         1960   
4  1954400510  20150218T000000  510000         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  sqft_above  sqft_basement   
0      5650     1.0           0     0  ...        1180              0  \
1      7242     2.0           0     0  ...        2170            400   
2     10000     1.0           0     0  ...         770              0   
3      5000     1.0           0     0  ...        1050            910   
4      8080     1.0           0     0  ...        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15   
0      1955             0    98178  47.5112 -122.257           1340  \
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  total_sqft  
0        5650       15000  
1        7639       21711  
2        8062       22322  
3        5000       15280  
4        7503       20743  

[5 rows x 22 columns]

 What will be the correlation between the columns sqft_living and sqft_above?
1. 0.702035 2. 0.754665 3. 0.876597 4. 0.303093

0.876597

In [147]:
np.corrcoef(df.sqft_above, df.sqft_living)[1,0]


0.876596598681318

36. If the correlation between the columns bathrooms and sqft_living is 0.754665, what all interpretations can be made about the two columns?
1. A positive correlation between the two columns
2. The columns show perfect correlation.
3. A strong negative correlation between the two columns
4. No correlation between the two columns

1.

In [150]:
np.corrcoef(df.bathrooms, df.sqft_living)[1,0]

0.7546652789673735